src/Compilers/VisualBasic/Portable/Binding/Binder_Lookup.vb

```diff
@@ -933,34 +930,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Public Shared Sub LookupDefaultProperty(result As LookupResult, container As TypeSymbol, binder As Binder, <[In], Out> ByRef useSiteDiagnostics As HashSet(Of DiagnosticInfo))
                 Select Case container.TypeKind
                     Case TypeKind.Class, TypeKind.Module, TypeKind.Structure
-                        Dim tempResult = LookupResult.GetInstance()
-                        LookupDefaultPropertyInClass(result, DirectCast(container, NamedTypeSymbol), binder, tempResult, useSiteDiagnostics)
-                        tempResult.Free()
+                        Dim tempResults = New TemporaryLookupResults(True)
```

Comment:
>TemporaryLookupResults [](start = 46, length = 22)

Consider just doing this once at the top of the method and the dispose at the bottom of the method, rather than in every case.

Reply:
I considered that, but it would mean doing it for anything that isn't hit by these cases.  i wasn't certain if that was an issue or not.  (i'd personally prefer to just move this to the caller).

Reply:
> I considered that, but it would mean doing it for anything that isn't hit by these cases.  i wasn't certain if that was an issue or not.  (i'd personally prefer to just move this to the caller).

If the structure would be fully lazy there would be no cost for code paths that do not use it.

---
In reply to: [361056713](https://github.com/dotnet/roslyn/pull/40522#discussion_r361056713) [](ancestors = 361056713)